core/chains/evm/logpoller/orm.go

```diff
@@ -400,6 +400,31 @@ func (o *ORM) SelectIndexLogsTopicGreaterThan(address common.Address, eventSig c
 	return logs, nil
 }
 
+func (o *ORM) SelectUntilBlockHashDataWordGreaterThan(address common.Address, eventSig common.Hash, wordIndex int, wordValueMin common.Hash, untilBlockHash common.Hash, qopts ...pg.QOpt) ([]Log, error) {
+	var logs []Log
+	q := o.q.WithOpts(qopts...)
+	err := q.Transaction(func(tx pg.Queryer) error {
+		// We want to mimic the behaviour of the ETH RPC which errors if blockhash not found.
+		var block LogPollerBlock
+		if err := tx.Get(&block,
+			`SELECT * FROM evm.log_poller_blocks 
+					WHERE evm_chain_id = $1 AND block_hash = $2`, utils.NewBig(o.chainID), untilBlockHash); err != nil {
```

Comment:
Interesting... this seems like a case where it would be very difficult to have a separate HeadTracker which is responsible for all block-related requests and a LogPoller component only responsible for logs.

If HeadTracker was the only place that stored blocks, you'd have to send a request to it to convert a block hash to a block number, but then there could be a re-org before the request for logs from that block reaches LogPoller.

(unrelated to this PR, but something I'll keep that in mind during those discussions.)